activerecord/test/cases/encryption/performance/key_provider_calls_test.rb

```diff
@@ -0,0 +1,48 @@
+# frozen_string_literal: true
+
+require "cases/encryption/helper"
+
+class ActiveRecord::Encryption::KeyProviderCallsTest < ActiveRecord::EncryptionTestCase
+  fixtures :encrypted_books, :posts
+
+  test "calls #encryption_key only once when creating a record" do
+    assert_equal 0, key_provider.encryption_key_calls
+
+    create_book
+
+    assert_equal 2, key_provider.encryption_key_calls
```

Comment:
At first glance, it's confusing that you're asserting two calls here but the test name says there's just one.

Reply:
If there is a good reason for that it might be worth adding a comment.